app/assets/javascripts/shopify_app/itp_polyfill.js

```diff
@@ -0,0 +1,16 @@
+function setCookieAndRedirect() {
```

Comment:
it may be wise to wrap all this code in an IIFE just for safety and to avoid interference from browser extensions, etc.